enhancements/ingress/customize-error-code-pages.md

```diff
@@ -0,0 +1,121 @@
+---
+title: customize-error-code-pages
+authors:
+  - "@miheer"
+reviewers:
+  - "@Miciah"
+  - "@danehans"
+  - "@knobunc"
+approvers:
+  - "@Miciah"
+  - "@danehans"
+  - "@knobunc"
+creation-date: 2021-03-07 
+last-updated: 2020-03-07
+status: implementable 
+see-also:
+replaces:
+superseded-by:
+---
+# Customize error code paged returned by the haproxy router
+## Release Signoff Checklist
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [X] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+## Summary
+There is no supported and reasonable method to customize the HAProxy 2.x error page as deployed in OpenShift.
+Customizing the HAProxy error page per our recommended procedure (documented here: https://access.redhat.com/solutions/3425711).
+does not work with OCP 4. It's also important that the right page as per user requirements is served at the right moment.
+Example - 503 error page shall be returned if no pod is available. 
+But when the URI is wrong/does not exist, we should return 404 or a customized 404 page as per user requirements.
+## Motivation
+In OCP 3.11 customizing error pages is possible via a [documented procedure](https://access.redhat.com/solutions/3425711)
+Customers resist customizing the HAProxy error page per our recommended procedure (documented here: https://access.redhat.com/solutions/3425711)
+Also this procedure is not possible in OCP 4.
+So while it is possible to define custom Error Pages, it's also important that the right page is served at the right moment. Currently 503 is 
+returned if no pod is available and also when the URI has a wrong target. But when the URI is wrong/does not
+exist, we should return 404 or something different as it otherwise may confuse end-customers.
+
+### Goals
+1. Enable the cluster administrator to specify custom error code pages for haproxy router to serve them at the right moment.
+
+### Non-Goal
+TBD
+
+## Proposal
+0. User creates a custom error code page configmap `http503page` in `openshift-config`
+1. The IngressController API is extended by adding an optional
+`HttpErrorCodePage` field with type string
+```go
+//This used to sync the configmap having customizable error code page created by cluster admin in openshift-config namespace.
```

Comment:
In general I would suggest using a single space after each comment indicator, ie `// <text>` instead of `//<text>`.

Reply:
OK